techniques/applications/repoGpgKeyManagement/1.0/repoGpgKeyManagement.st

```diff
@@ -0,0 +1,135 @@
+      classes    => if_else("repo_gpg_key_${index_repo_gpg_keys}_removed", "repo_gpg_key_${index_repo_gpg_keys}_remove_error");
+
+    "/bin/rpm --quiet --import ${g.rudder_var_tmp}/gpg_key_${index_repo_gpg_keys}_content"
+      comment    => "This imports the GPG key if the necessary tmp file has been created and it needs to be imported and is not present",
+      ifvarclass => "repo_gpg_key_${index_repo_gpg_keys}_action_add.!repo_gpg_key_${repo_gpg_hash_lc[${index_repo_gpg_keys}]}_present.gpg_key_file_${index_repo_gpg_keys}_created",
+      classes    => if_else("repo_gpg_key_${index_repo_gpg_keys}_imported", "repo_gpg_key_${index_repo_gpg_keys}_import_error");
+
+  reports:
+    
+    "@@repoGpgKeyManagement@@result_repaired@@${repo_gpg_uuid[${index_repo_gpg_keys}]}@@GPG Key Management@@${repo_gpg_hash[${index_repo_gpg_keys}]}@@${g.execRun}##${g.uuid}@#The GPG Key was imported successfully" 
+      ifvarclass => "repo_gpg_key_${index_repo_gpg_keys}_imported";
+
+    "@@repoGpgKeyManagement@@result_success@@${repo_gpg_uuid[${index_repo_gpg_keys}]}@@GPG Key Management@@${repo_gpg_hash[${index_repo_gpg_keys}]}@@${g.execRun}##${g.uuid}@#The GPG Key is already imported" 
+      ifvarclass => "repo_gpg_key_${index_repo_gpg_keys}_action_add.repo_gpg_key_${repo_gpg_hash_lc[${index_repo_gpg_keys}]}_present";
+
+    "@@repoGpgKeyManagement@@result_error@@${repo_gpg_uuid[${index_repo_gpg_keys}]}@@GPG Key Management@@${repo_gpg_hash[${index_repo_gpg_keys}]}@@${g.execRun}##${g.uuid}@#The GPG Key could not be imported"
+      ifvarclass => "repo_gpg_key_${index_repo_gpg_keys}_import_error";
+
+    "@@repoGpgKeyManagement@@result_repaired@@${repo_gpg_uuid[${index_repo_gpg_keys}]}@@GPG Key Management@@${repo_gpg_hash[${index_repo_gpg_keys}]}@@${g.execRun}##${g.uuid}@#The GPG Key was removed successfully"
+      ifvarclass => "repo_gpg_key_${index_repo_gpg_keys}_removed"; 
+
+    "@@repoGpgKeyManagement@@result_success@@${repo_gpg_uuid[${index_repo_gpg_keys}]}@@GPG Key Management@@${repo_gpg_hash[${index_repo_gpg_keys}]}@@${g.execRun}##${g.uuid}@#The GPG Key is not imported"
+      ifvarclass => "repo_gpg_key_${index_repo_gpg_keys}_action_remove.!repo_gpg_key_${repo_gpg_hash_lc[${index_repo_gpg_keys}]}_present";
+
+    "@@repoGpgKeyManagement@@result_error@@${repo_gpg_uuid[${index_repo_gpg_keys}]}@@GPG Key Management@@${repo_gpg_hash[${index_repo_gpg_keys}]}@@${g.execRun}##${g.uuid}@#The GPG Key could not be removed"
+      ifvarclass => "repo_gpg_key_${index_repo_gpg_keys}_remove_error";
+
+    "@@repoGpgKeyManagement@@result_error@@${repo_gpg_uuid[${index_repo_gpg_keys}]}@@GPG Key Management@@${repo_gpg_hash[${index_repo_gpg_keys}]}@@${g.execRun}##${g.uuid}@#The temporary file could not be created"
+      ifvarclass => "gpg_key_file_${index_repo_gpg_keys}_create_failed";
+
+}
+
+bundle edit_line set_gpg_file_content(c)
+{
+  insert_lines:
+    "${c}";
+}
+
+body edit_defaults empty_nobackup
```

Comment:
i'm amazed that we don't already have this one on the stdlib. See my comment up there